src/frontend/src/icp/components/transactions/IcTransactions.svelte

```diff
@@ -52,6 +54,9 @@
 
 	let noTransactions = false;
 	$: noTransactions = nonNullish($token) && $icTransactionsStore?.[$token.id] === null;
+
+	let hasIndexCanister = false;
```

Comment:
Isn't `isIcTokenCanistersStrict($tokenAsIcToken ?? ICP_TOKEN)` returning `true` per default which is the contrary of `hasIndexCanister = false;` -> should it be `hasIndexCanister = true;`

Reply:
ah, you mean for the fallback `ICP_TOKEN`, right? I will add a nullish check before

Reply:
In general, I think the token should be defined here alway, but we still use the token store

Reply:
No I mean, if `$tokenAsIcToken === null` then it's `ICP_TOKEN` that is used and `isIcTokenCanistersStrict(ICP_TOKEN)` equals `true` so when the component is mounted without the `tokenAsIcToken` we got true fals

```
let hasIndexCanister = false; // <---- False on mount
hasIndexCanister = isIcTokenCanistersStrict($tokenAsIcToken ?? ICP_TOKEN) // <---- True if undefined on mount
```

e.g. shouldnt it be

```
let hasIndexCanister = true; // <---- true 
hasIndexCanister = isIcTokenCanistersStrict($tokenAsIcToken ?? ICP_TOKEN) // <---- True
```

But maybe I'm overseeing something?

Reply:
yes yes! i had understood exactly that! i fixed it with the `nonNullish` before and removing `ICP_TOKEN`? Is that ok?

Reply:
now I did 😉